torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -1423,14 +1423,16 @@ std::vector<c10::Device> TensorPipeAgent::getDevicesForRemote(
   if (iter == deviceMaps.end()) {
     for (const auto& t : message.tensors()) {
       TORCH_CHECK(
-          t.device().is_cpu(),
+          t.device().is_cpu() || t.device().is_meta(),
           errStr,
           ", but found tensor on device: ",
           t.device());
     }
-    return {};
+    return getDevicesForTensors(message.tensors(), {{c10::kCPU, c10::kCPU}, {c10::kMeta, c10::kMeta}}, errStr);
```

Comment:
do we need to add checks somewhere that meta -> not-meta or not-meta -> meta is an error?

Reply:
do we have to specify cpu and meta here? Or will it work if we default to {CPU->CPU, Meta->Meta} if device map is not specified?

Reply:
reverted